GUI/src/cz/fidentis/gui/actions/landmarks/EditLandmarkID.java

```diff
@@ -144,25 +158,31 @@ private void cancelButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-
 
     private void okButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_okButtonActionPerformed
        int id = (Integer)landmarkIDSpinner.getValue();
-        boolean contains = info.containsFP(id);
+       boolean contains = info.containsFP(id);
        
-       if(contains){
+       if(id != fp.getType() && contains){
            int result = JOptionPane.showConfirmDialog(this, "Landmark with this ID already exists. Do you want to rewrite it?", "Landmark exists", JOptionPane.YES_NO_OPTION);
            
            if(result == JOptionPane.YES_OPTION){
                info.removeFacialPoint(id);
-                setFPInfo(id);
+               setFPInfo(id);
                this.dispose();
            }
        }else{
-            setFPInfo(id);
+           setFPInfo(id);
            this.dispose();
        }         
     }//GEN-LAST:event_okButtonActionPerformed
 
+    private void toggleActiveButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_toggleActiveButtonActionPerformed
+        //only change state locally, fp state will change if user clicks OK button
```

Comment:
👍  chvalim, ja by som si to na prvy pohlad neuvedomil ze to treba spravit najprv lokalne a ulozit az ked sa to potvrdi :)